### PR TITLE
Add ipsrc for rspconfig, fix rspconfig vlan reading

### PR DESCRIPTION
**Pull request by zet809**
To replace pull request #3843 
1. add option ipsrc for rspconfig
2. fix rspconfig vlan reading

The output looks like this:
```
[root@briggs01 xCAT_plugin]# XCAT_OPENBMC_DEVEL=YES rspconfig mid05tor12cn13,mid05tor12cn05 netmask gateway vlan ip ipsrc
mid05tor12cn13: BMC Netmask: 255.255.0.0
mid05tor12cn13: BMC Gateway: 0.0.0.0 (default: n/a)
mid05tor12cn13: BMC VLAN ID: 11
mid05tor12cn13: BMC IP: 172.11.139.13
mid05tor12cn13: BMC IP Source: Static
mid05tor12cn05: BMC Netmask: 255.255.0.0
mid05tor12cn05: BMC Gateway: 0.0.0.0 (default: 172.12.253.27)
mid05tor12cn05: BMC VLAN ID: Disabled
mid05tor12cn05: BMC IP: 172.12.253.100
mid05tor12cn05: BMC IP Source: DHCP
```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -630,6 +630,7 @@ sub parse_args {
                 return ([ 1, "Can not configure and display nodes' value at the same time" ]) if ($setorget and $setorget eq "get");
                 my $key = $1;
                 my $value = $2;
+                return ([ 1, "Unsupport to configure ipsrc" ]) if ($key eq "ipsrc");
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Suggest changing to `Changing ipsrc value is currently not supported.`

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 2]
```diff
@@ -1838,11 +1840,14 @@ sub rspconfig_response {
...
 
-            ($path, $adapter_id) = (split(/ipv4\//, $key_url));
-
+            ($path, $adapter_id) = (split(/\/ipv4\//, $key_url));
+            
             if ($adapter_id) {
                 if (defined($content{Address}) and $content{Address}) {
                     unless ($address =~ /n\/a/) {
+                        if ($ipsrc eq "DHCP" and $xcatdebugmode) {
+                            process_debug_info($node, "There is an IP address:$address got through DHCP");
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Suggest change `got` to `obtained`

File: xCAT-client/pods/man1/rspconfig.1.pod [hunk 3]
```diff
@@ -308,6 +308,10 @@ Get or set hostname on the service processor.
 
 Get or set vlan ID. For get vlan ID, if vlan is not enabled, 'BMC VLAN disabled' will be outputed. For set vlan ID, the valid value are [1-4096].
 
+=item B<ipsrc>
+
+Get the ip source for OpenBMC.
```
Comment (whowutwut) [inline xCAT-client/pods/man1/rspconfig.1.pod]: Suggest changing ip -> IP in caps for the descriptions, (NOT the options)...   there are another references in this file just below at line 317.

**Comment by zet809** [comment]
Hi, @whowutwut , modified, pls review again. Thx!

**Pull request merged**